acceptance/support/application-manager.ts

```diff
@@ -0,0 +1,88 @@
+import * as fs from "fs";
+import {Logger} from "tslog";
+import {uploadPath} from "./init";
+import {TestData} from "./test-constants";
+
+/**
+ * Utilities for managing LSRS while executing the Cucumber acceptance tests.
+ */
+export default class ApplicationManager {
+
+    private static readonly startUpTimeout = 5000;
+    private static readonly logger: Logger = new Logger({
+        name: "Cucumber | ApplicationManager",
+        minLevel: "debug",
+        displayFunctionName: false,
+        displayFilePath: "hidden"
+    });
+
+    private static started: boolean = false;
+
+    /**
+     * Starts up LSRS. Returns immediately if the application is already running.
+     */
+    public static async start(): Promise<void> {
+
+        return new Promise((resolve) => {
+
+            if (this.started) {
+                this.notifyStarted(resolve, "Application is already loaded");
+            } else {
+
+                this.logger.info(`Starting application with environment=${process.env.NODE_ENV}...`);
+
+                this.started = true;
+                import("../../src/lsrs-main");
```

Comment:
You could consider exposing some `main` function in `lsrs-main` that can be called here. Inside the file, you can check if the module is run directly or imported, and if the former, you can call the main function there.

This could also make it simpler to determine if the app is actually running.

Reply:
I've got a "policy" of mine about not adding any piece of test-support code into the live code base. So I'd rather stick to a workaround like this, even if it feels hacky. :/

Reply:
That is true, *however*. I generally think it is a good practice to have a sort of main function, just so that the application entry is not an import with a side effect, but an actual method.
Granted I'm not sure how widespread this practice is ( outside of C/C++ 😄 ), but I wouldn't consider it as test support code.

Reply:
Jokes on you, **there is** a main. In the leaflet-static-resource-server-application.ts module. :D